snuba/web/views.py

```diff
@@ -284,9 +306,7 @@ def dataset_query(dataset: Dataset, body, timer: Timer) -> Response:
             dataset.get_extensions(), HTTPRequestSettings
         )
 
-    request = validate_request_content(
-        body, schema, timer, dataset, http_request.referrer,
-    )
+    request = build_request(body, schema, timer, dataset, http_request.referrer)
 
     try:
         result = parse_and_run_query(dataset, request, timer)
```

Comment:
@tkaemming do you recall why we manage `QueryException` inline in this method instead of using an error handler like we do for a few other exceptions ?

Reply:
I don't recall any specific reason, I think it may have just never been updated to use the error handlers. (I wouldn't be surprised if there are some tests that depend on this returning responses from this function instead of throwing but I didn't actually look.)